File: generate_markdown.py
import json

with open('/tmp/benchmark-results.json', 'r') as f:
    data = json.load(f)

def fmt_time(v):
    return 'N/A' if v < 0 else f"{v:.1f}s"

results_table = f"""| Metric | Value |
|--------|-------|
| Scale-up time | {fmt_time(data['scaleUpTimeSec'])} |
| Scale-down time | {fmt_time(data['scaleDownTimeSec'])} |
| Max replicas | {data['maxReplicas']} |
| Avg KV cache usage | {data['avgKVCacheUsage']:.3f} |
| Avg queue depth | {data['avgQueueDepth']:.1f} |
| Replica oscillation (σ) | {data['replicaOscillation']:.2f} |
| Total duration | {data['totalDurationSec']:.0f}s |"""

print("### Benchmark: scale-up-latency (Kind)\n")
print(results_table)

File: generate_markdown.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('/tmp/benchmark-results.json', 'utf8'));
const fmtTime = (v) => v < 0 ? 'N/A' : `${v.toFixed(1)}s`;

let resultsTable = `| Metric | Value |
|--------|-------|
| Scale-up time | ${fmtTime(data.scaleUpTimeSec)} |
| Scale-down time | ${fmtTime(data.scaleDownTimeSec)} |
| Max replicas | ${data.maxReplicas} |
| Avg KV cache usage | ${data.avgKVCacheUsage.toFixed(3)} |
| Avg queue depth | ${data.avgQueueDepth.toFixed(1)} |
| Replica oscillation (σ) | ${data.replicaOscillation.toFixed(2)} |
| Total duration | ${data.totalDurationSec.toFixed(0)}s |`;

console.log("### Benchmark: scale-up-latency (Kind)\n");
console.log(resultsTable);
